cpp/src/link_analysis/pagerank_impl.cuh

```diff
@@ -151,8 +140,8 @@ void pagerank(
         pull_graph_view.local_vertex_partition_view());
       auto num_invalid_vertices =
         thrust::count_if(handle.get_thrust_policy(),
-                         *personalization_vertices,
-                         *personalization_vertices + *personalization_vector_size,
+                         std::get<0>(*personalization).begin(),
+                         std::get<0>(*personalization).end(),
```

Comment:
Should there be an additional check to make sure that the vertices provided in personalization are unique? It would cause any problem if they are not unique though, but the last value (for a particular vertex) would be used.

Reply:
Actually, I think if a vertex is specified multiple times, all values would be used.  We iterate over the `(v, value)` pairs and add in the personalization effect to the pagerank for vertex `v`.  So if it is specified multiple times that accumulation will occur multiple times.  But that update occurs non-atomically, so `v` being specified multiple times would result in non-deterministic behavior.

Added a check in the next push